scipy/stats/tests/test_morestats.py

```diff
@@ -1658,22 +1674,26 @@ def test_exact_p_1(self, x):
         assert_equal(p, 1)
 
     def test_auto(self):
-        # auto default to exact if there are no ties and n<= 25
-        x = np.arange(0, 25) + 0.5
-        y = np.arange(25, 0, -1)
+        # auto default to exact if there are no ties and n <= 50
+        x = np.arange(0, 50) + 0.5
+        y = np.arange(50, 0, -1)
         assert_equal(stats.wilcoxon(x, y),
                      stats.wilcoxon(x, y, mode="exact"))
 
-        # if there are ties (i.e. zeros in d = x-y), then switch to approx
+        # n <= 50: if there are zeros in d = x-y, use PermutationMethod
+        pm = stats.PermutationMethod()
         d = np.arange(0, 13)
-        with suppress_warnings() as sup:
-            sup.filter(UserWarning, message="Exact p-value calculation")
-            w, p = stats.wilcoxon(d)
-        assert_equal(stats.wilcoxon(d, mode="approx"), (w, p))
+        w, p = stats.wilcoxon(d)
+        assert_equal(stats.wilcoxon(d, method=pm), (w, p))
```

Comment:
These are `np.testing.assert_equal`, so it doesn't matter like it does for `np.testing.assert_allclose` or `xp_assert_equal`, but please order the arguments with the result being tested first and the reference second. ```suggestion
        assert_equal((w, p), stats.wilcoxon(d, method=pm))
```